test/examples/test_examples.py

```diff
@@ -25,15 +61,100 @@ def test_basic_configuration():
                 received += 1
                 continue
 
-        if sent * 2 == received:
+        if sent != 0 and received != 0 and sent * 2 == received:
+            ret = True
+        else:
+            print ('ERROR: sent: ' + str(sent) + ', but received: ' + str(received) + ' (expected: ' + str(sent * 2) + ')')
+            raise subprocess.CalledProcessError(1, out.decode())
+
+    except subprocess.CalledProcessError as e:
+        print (e.output)
+        #for l in e.output.split('\n'):
+        #    print(l)
+    except subprocess.TimeoutExpired:
+        print('TIMEOUT')
+        print(out)
+
+    assert(ret)
+
+timeout_test_cases = [
+    ('--transport SHM', '--transport UDPv4'),                                     # Builtin transports
+    ('--deadline 300', '--deadline 100'),                                         # Deadline QoS
+    ('', '--disable-positive-ack'),                                                # Disable positive ACKs QoS
+    ('', '--transient-local'),                                                     # Durability QoS
+    #('--keep-last 10 --max-samples-per-instance 1', ''), # this one only displays warning from now on
+    ('--liveliness-kind AUTOMATIC', '--liveliness-kind MANUAL_BY_PARTICIPANT'),   # Liveliness QoS
+    ('--liveliness-kind AUTOMATIC', '--liveliness-kind MANUAL_BY_TOPIC'),
+    ('--liveliness-kind MANUAL_BY_PARTICIPANT', '--liveliness-kind MANUAL_BY_TOPIC'),
+    ('--ownership', ''),                                                           # Ownership QoS
+    ('', '--ownership'),
+    ('--partition configuration_example_partition', '--partition other_partition'), # Partition QoS
+    ('', '--reliable'),                                                          # Reliability QoS
+]
+
+@pytest.mark.parametrize("pub_args, sub_args", timeout_test_cases)
+def test_configuration_timeout(pub_args, sub_args):
+    """."""
+    ret = False
+    out = ''
+
+    command_prerequisites = 'PUB_ARGS="' + pub_args + '" SUB_ARGS="' + sub_args + '" '
+    try:
+        out = subprocess.check_output(command_prerequisites + '@DOCKER_EXECUTABLE@ compose -f configuration.compose.yml up',
+            stderr=subprocess.STDOUT,
+            shell=True,
+            timeout=10
+        )
+    except subprocess.CalledProcessError as e:
+        print (e.output)
+    except subprocess.TimeoutExpired:
+        ret = True
+        subprocess.check_output('@DOCKER_EXECUTABLE@ compose -f configuration.compose.yml down',
+            stderr=subprocess.STDOUT,
+            shell=True,
+            timeout=15
+        )
+
+    assert(ret)
+
+expected_output_test_cases = [
```

Comment:
I think it would be nice adding some output test case for the liveliness changed/lost